corehq/form_processor/abstract_models.py

```diff
@@ -215,6 +215,14 @@ def to_api_json(self):
     def set_case_id(self, case_id):
         raise NotImplementedError()
 
+    @property
+    def server_opened_on(self):
+        try:
+            open_action = self.actions[0]
```

Comment:
One reason not to do this is that it will load all the case transactions from the DB